src/main/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentMap.java

```diff
@@ -298,8 +300,11 @@ public void run() {
 
         private final int hashCode;
 
-        WeakKey(T key, ReferenceQueue<? super T> queue) {
+        private T key;
```

Comment:
This creates a strong reference to the key, thus it's not actually a `WeakReference` anymore.

Reply:
Thanks for pointing it !!